src/main.js

```diff
@@ -130,92 +99,26 @@ let load = loader({
   },
 
   app: {
-    requires: ['cfg', 'authenticators', 'router'],
-    setup: ({cfg, authenticators, router}) => {
+    requires: ['cfg', 'docs', 'router'],
+    setup: ({cfg, docs, router}) => {
       // Create application
-      let app = tcApp(_.defaults({}, cfg.server, {contentSecurityPolic: false}));
-
-      // setup 'trust proxy', which tc-lib-app does not do
-      app.set('trust proxy', cfg.server.trustProxy);
-
-      // Setup API
-      app.use('/v1', router);
-
-      // Setup views and assets
-      app.use('/assets', express.static(path.join(__dirname, '..', 'assets')));
-      app.set('views', path.join(__dirname, '..', 'views'));
-      app.set('view engine', 'jade');
-
-      // build a CSP that's appropriate to this app..
-      app.use(csp.getCSP({
-        'default-src': csp.SRC_NONE,
-        'frame-ancestors': csp.SRC_NONE,
-        'base-uri': csp.SRC_NONE,
-        'report-uri': '/__cspreport__',
-        'font-src': csp.SRC_SELF,
-        'script-src': [csp.SRC_SELF, '\'unsafe-inline\''],
-        'style-src': [csp.SRC_SELF, '\'unsafe-inline\''],
-      }));
-
-      // Parse request bodies
-      app.use(bodyParser.urlencoded({extended: false}));
-
-      // Store session in a signed cookie
-      app.use(session({
-        name: 'taskcluster-login',
-        keys: cfg.app.cookieSecrets,
-        secure: cfg.server.forceSSL,
-        secureProxy: cfg.server.trustProxy,
-        httpOnly: true,
-        signed: true,
-        maxAge: 5 * 60 * 1000,
-      }));
-
-      // Set up message flashing
-      app.use(flash());
-
-      // Initialize passport
-      app.use(passport.initialize());
-      app.use(passport.session());
-
-      // Read and write user from signed cookie
-      passport.serializeUser((user, done) => done(null, user.serialize()));
-      passport.deserializeUser((data, done) => done(null, User.deserialize(data)));
-
-      // set up authenticators' sub-paths
-      _.forIn(authenticators, (authn, name) => {
-        app.use('/' + name, authn.router());
-      });
-
-      // Add logout method
-      app.post('/logout', (req, res) => {
-        req.logout();
-        res.redirect('/');
-      });
-
-      // Render index
-      app.get('/', (req, res) => {
-        let user = User.get(req);
-        let {credentials} = user.createCredentials(cfg.app.temporaryCredentials);
-        res.render('index', {
-          user, credentials,
-          querystring,
-          allowedHosts: cfg.app.allowedRedirectHosts,
-          query: req.query,
-          flash: req.flash(),
-          session: req.session,
-          auth0_domain: cfg.auth0.domain,
-          auth0_client_id: cfg.auth0.clientId,
-        });
+      let app = App({
+        port: cfg.server.port,
+        publicUrl: cfg.server.publicUrl,
+        env: cfg.server.env,
+        forceSSL: cfg.server.forceSSL,
+        trustProxy: cfg.server.trustProxy,
+        rootDocsLink: true, // doesn't work?
```

Comment:
It doesn't work?

Reply:
it does -- I've removed the comment now.